config/main.py

```diff
@@ -1635,19 +1630,23 @@ def load_mgmt_config(filename):
         mgmt_conf = netaddr.IPNetwork(key[1])
         gw_addr = config_data['MGMT_INTERFACE'][key]['gwaddr']
         if mgmt_conf.version == 4:
-            command = "ifconfig eth0 {} netmask {}".format(str(mgmt_conf.ip), str(mgmt_conf.netmask))
+            command = ['ifconfig', 'eth0', str(mgmt_conf.ip), 'netmask', str(mgmt_conf.netmask)]
             clicommon.run_command(command, display_cmd=True)
         else:
-            command = "ifconfig eth0 add {}".format(str(mgmt_conf))
+            command = ['ifconfig', 'eth0', 'add', str(mgmt_conf)]
             # Ignore error for IPv6 configuration command due to it not allows config the same IP twice
             clicommon.run_command(command, display_cmd=True, ignore_error=True)
-        command = "ip{} route add default via {} dev eth0 table default".format(" -6" if mgmt_conf.version == 6 else "", gw_addr)
+        command = ['ip{}'.format(" -6" if mgmt_conf.version == 6 else ""), 'route', 'add', 'default', 'via', str(gw_addr), 'dev', 'eth0', 'table', 'default']
         clicommon.run_command(command, display_cmd=True, ignore_error=True)
-        command = "ip{} rule add from {} table default".format(" -6" if mgmt_conf.version == 6 else "", str(mgmt_conf.ip))
+        command = ['ip{}'.format(" -6" if mgmt_conf.version == 6 else ""), 'rule', 'add', 'from', str(mgmt_conf.ip), 'table', 'default']
         clicommon.run_command(command, display_cmd=True, ignore_error=True)
     if len(config_data['MGMT_INTERFACE'].keys()) > 0:
-        command = "[ -f /var/run/dhclient.eth0.pid ] && kill `cat /var/run/dhclient.eth0.pid` && rm -f /var/run/dhclient.eth0.pid"
-        clicommon.run_command(command, display_cmd=True, ignore_error=True)
+        pid_cmd = ['cat', '/var/run/dhclient.eth0.pid']
+        _, pid = getstatusoutput_noshell(pid_cmd)
+        cmd0 = ['[',  '-f', '/var/run/dhclient.eth0.pid', ']']
```

Comment:
>'-f' [](http://example.com/codeflow?start=22&length=4)

`-f` is a bash/sh feature, I guess you need to run_command with `shell=True` to have this feature. Could you check? If yes, you may consider using python to check, like `os.path.isfile(path)`.